src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ExpressionHelper.cs

```diff
@@ -163,5 +188,37 @@ public static bool IsSingleArgumentIndexer(Expression expression)
                 property => (string.Equals(defaultMember.MemberName, property.Name, StringComparison.Ordinal) &&
                     property.GetMethod == methodExpression.Method));
         }
+
+        public static bool EqualsTo(this Expression expression1, Expression expression2)
```

Comment:
Why does this exist? And why is it an extension method and in this class? Inline the useful parts into `LambdaExpressionComparer`.

Note the Engineering Guidelines again: We define extension methods in a few special cases, none where a `static` method would be sufficient.